app/models/vm.rb

```diff
@@ -45,35 +45,14 @@ def enforce_single_service_parent?
   end
 
   def self.find_all_by_mac_address_and_hostname_and_ipaddress(mac_address, hostname, ipaddress)
-    return [] if mac_address.blank? && hostname.blank? && ipaddress.blank?
-
-    include = [:vm_or_template]
-    references = []
-    conds = [["hardwares.vm_or_template_id IS NOT NULL"]]
-    if mac_address
-      conds[0] << "guest_devices.address = ?"
-      conds << mac_address
-      include << :nics
-      references << :guest_devices
-    end
-    if hostname
-      conds[0] << "networks.hostname = ?"
-      conds << hostname
-      include << :networks
-      references << :networks
-    end
-    if ipaddress
-      conds[0] << "networks.ipaddress = ?"
-      conds << ipaddress
-      include << :networks
-      references << :networks
-    end
-    conds[0] = "(#{conds[0].join(" AND ")})"
-
-    Hardware.includes(include.uniq)
-            .references(references.uniq)
-            .where(conds)
-            .map(&:vm_or_template).select { |vm| vm.kind_of?(Vm) }
+    return Vm.none if mac_address.blank? && hostname.blank? && ipaddress.blank?
+    scope = Vm
```

Comment:
Minor rubocop, but there should be a blank line after a guard clause


```suggestion
    return Vm.none if mac_address.blank? && hostname.blank? && ipaddress.blank?

    scope = Vm
```